src/primitives/transaction.h

```diff
@@ -200,6 +200,51 @@ class CTxOut
 extern boost::atomic<bool> flexTransActive;
 struct CMutableTransaction;
 
+template<typename Stream>
+void SerialiseScriptSig4(const std::vector<CTxIn> &inputs, Stream &s, int nType, int nVersion)
+{
+    for (auto in : inputs) {
+        bool first = true;
+        unsigned int i = 0; // use both iterator and int b/c the iterator doesn't do bounds-checks.
+        auto iter = in.scriptSig.begin();
+        while (i < in.scriptSig.size()) {
+            uint8_t k = *iter;
+            if ((k > 0 && k < 76) || (k >= 76 && k <= 78)) {
```

Comment:
So you actually decode as a series of push, and then rencode by decoding the pushes. Sure it should work, but I'm not sure this is the best design :)

One of the cool thing about having a stack description is that you can just use it and interpret the out script, no need to interpret the first part.